### PR TITLE
M3-3335 Skeleton Component

### PR DESCRIPTION
**Pull request by alioso**
## Skeleton Component

This replaces the previously failed iteration of the skeleton component and uses the MUI (lab) version of this component. Right now this is only used for tablesRowLoading. If using for other components it will probably need new props and some finer logic.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers
- run `yarn` to get new deps

### CODE REVIEW

File: packages/manager/src/components/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    columns,
+    firstColWidth,
+    variant,
+    textHeight,
+    subtextHeight
+  } = props;
+
+  const cols: any = [];
+  const renderTableSkeleton = (colCount: number) => {
+    const ifColumns = columns !== undefined ? columns : 1;
```
Comment (Jskobos) [inline packages/manager/src/components/Skeleton/Skeleton.tsx]: You do this calc or something similar a few times, you can just lift this statement up out of the function, then use `ifColumns` (or `_columns`) or whatever throughout the rest of the component.

File: packages/manager/src/components/Skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        return firstColWidth ? firstColWidth : 100 / ifColumns;
+      } else {
+        return firstColWidth
+          ? (100 - firstColWidth) / (ifColumns - 1)
+          : 100 / ifColumns;
+      }
+    };
+    for (
+      colCount = 0;
+      colCount <= (columns !== undefined ? columns - 1 : 1);
```
[the remark this answers is not in the document]
Comment (alioso) [reply packages/manager/src/components/Skeleton/Skeleton.tsx]: all set

File: packages/manager/src/components/Skeleton/Skeleton.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+              />
+            </Grid>
+          </Grid>
+        </Grid>
+      );
+    }
+    return;
+  };
+
+  table && columns !== undefined
```
[the remark this answers is not in the document]
Comment (alioso) [reply packages/manager/src/components/Skeleton/Skeleton.tsx]: donzzz

File: packages/manager/src/components/Skeleton/Skeleton.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    ? renderTableSkeleton(columns)
+    : renderTableSkeleton(1);
+
+  return (
+    <>
+      {table && columns !== undefined ? (
```
[the remark this answers is not in the document]
Comment (alioso) [reply packages/manager/src/components/Skeleton/Skeleton.tsx]: true, streamlined

File: packages/manager/src/components/Skeleton/Skeleton.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  const {
+    table,
+    columns,
+    firstColWidth,
+    variant,
+    textHeight,
+    subtextHeight
+  } = props;
+
+  const cols: any = [];
```
[the remark this answers is not in the document]
Comment (alioso) [reply packages/manager/src/components/Skeleton/Skeleton.tsx]: addressed

File: packages/manager/src/components/Skeleton/Skeleton.tsx (hunk 1 above)
Comment (alioso) [reply packages/manager/src/components/Skeleton/Skeleton.tsx]: done

**Pull request merged**